include/cuco/static_map.cuh

```diff
@@ -181,6 +177,13 @@ class static_map {
   using slot_allocator_type =
     typename std::allocator_traits<Allocator>::rebind_alloc<pair_atomic_type>;
 
+#if defined(__CUDA_ARCH__) && (__CUDA_ARCH__ < 700)
+  static_assert(atomic_key_type::is_always_lock_free,
```

Comment:
Why do these two checks do / why are they in place?

Reply:
It would fail to compile either way because `cuda::atomic<T>` would complain about >8B on < sm_70, but this just provides a more helpful error message.